realtime/setupSocketIO.js

```diff
@@ -68,23 +68,25 @@ function setupNamespace(io) {
     .then((objArr) => {
       if (objArr) {
         objArr.forEach((o) => rtUtils.initializePerspectiveNamespace(o, io));
-        room.findAll()
-        .then((rooms) => {
-          if (rooms) {
-            rooms.forEach((r) => rtUtils.initializeBotNamespace(r, io));
-            resolve(io);
-          } else {
-            const err = new ResourceNotFoundError();
-            err.resourceType = 'Room';
-            throw err;
-          }
-        });
       } else {
         const err = new ResourceNotFoundError();
         err.resourceType = 'Perspective';
         throw err;
       }
     })
+    .then(() => {
+      room.findAll()
+      .then((rooms) => {
+        if (rooms) {
+          rooms.forEach((r) => rtUtils.initializeBotNamespace(r, io));
+          resolve(io);
+        } else {
+          const err = new ResourceNotFoundError();
+          err.resourceType = 'Room';
+          throw err;
+        }
+      });
+    })
```

Comment:
probably flatten this promise?
`.then(() => room.findAll())
.then(() => { if(rooms) ......})`

Reply:
Can you remove the else blocks too, 83 to 87 and 71 to 75. I think we will never hit these lines.